opencodeblocks/scene/ipynb_conversion.py

```diff
@@ -0,0 +1,241 @@
+
+    adujst_markdown_blocks_width(blocks)
+
+    return blocks
+
+
+def get_default_block() -> OrderedDict:
+    """Return a default block with argument that vary missing"""
+    return {
+        "title": "_",
+        "splitter_pos": [
+            85,
+            261,
+        ],
+        "sockets": [],
+        "metadata": {
+            "title_metadata": {
+                "color": "white",
+                "font": "Ubuntu",
+                "size": 12,
+                "padding": 4.0,
+            }
+        },
+    }
+
+
+def is_title(block: OrderedDict) -> bool:
+    """Checks if the block is a one-line markdown block which could correspond to a title"""
+
+    if block["block_type"] != BLOCK_TYPE_TO_NAME["markdown"]:
+        return False
+    if "\n" in block["text"]:
+        return False
+    if len(block["text"]) == 0 or len(block["text"]) > TITLE_MAX_LENGTH:
+        return False
+    # Headings, quotes, bold or italic text are not considered to be headings
+    if block["text"][0] in {"#", "*", "`"}:
+        return False
+    return True
+
+
+def adujst_markdown_blocks_width(blocks: OrderedDict) -> None:
+    """Modify the markdown blocks width (in place) for them to match the width of block of code below"""
+    i: int = len(blocks) - 1
+
+    while i >= 0:
+        if blocks[i]["block_type"] == BLOCK_TYPE_TO_NAME["code"]:
+            block_width: float = blocks[i]["width"]
+            i -= 1
+
+            while i >= 0 and blocks[i]["block_type"] == BLOCK_TYPE_TO_NAME["markdown"]:
+                blocks[i]["width"] = block_width
+                i -= 1
+        else:
+            i -= 1
+
+
+def add_sockets(blocks: OrderedDict) -> OrderedDict:
+    """Add sockets to the blocks (in place) and returns the edge list"""
+    code_blocks: List[OrderedDict] = [
+        block for block in blocks if block["block_type"] == BLOCK_TYPE_TO_NAME["code"]
+    ]
+    edges: List[OrderedDict] = []
+
+    for i in range(1, len(code_blocks)):
+        socket_id_out = len(blocks) + 2 * i
+        socket_id_in = len(blocks) + 2 * i + 1
+        code_blocks[i - 1]["sockets"].append(
+            get_default_output_socket(socket_id_out, code_blocks[i - 1]["width"])
+        )
+        code_blocks[i]["sockets"].append(get_default_input_socket(socket_id_in))
+        edges.append(
+            get_default_edge(
+                i,
+                code_blocks[i - 1]["id"],
+                socket_id_out,
+                code_blocks[i]["id"],
+                socket_id_in,
+            )
+        )
+    return edges
+
+
+def get_default_input_socket(socket_id: int) -> OrderedDict:
+    """Returns the default input socket with the corresponding id"""
+    return {
+        "id": socket_id,
+        "type": "input",
+        "position": [0.0, SOCKET_HEIGHT],
+        "metadata": {
+            "color": "#FF55FFF0",
+            "linecolor": "#FF000000",
+            "linewidth": 1.0,
+            "radius": 10.0,
+        },
+    }
```

Comment:
Should not add random metadata, deserialization should handle missing arguments with default ones. If that's not the case, raise an issue and start a bugfix branch from master.
```suggestion
def get_default_input_socket(socket_id: int) -> OrderedDict:
    """Returns the default input socket with the corresponding id"""
    return {"id": socket_id, "type": "input"}
```